source/material_model/morency_doin.cc

```diff
@@ -93,7 +93,7 @@ namespace aspect
             for (unsigned int j=0; j < volume_fractions.size(); ++j)
               np += volume_fractions[j] * nps[j];
 
-            const double tauy = tau_0 + gamma*reference_density()*g*depth;
+            const double tauy = tau_0 + gamma*densities[0]*g*depth;
```

Comment:
Is this going to break anything? Or does the material model use a constant density anyway?

If the formula really requires a reference density, then it ought to become a parameter of the material model.